### PR TITLE
Implement reruns functionality and fix data directory parameter

### PR DESCRIPTION
**Pull request by anastasiyaig**
Partially addresses https://github.com/status-im/desktop-qa-automation/issues/558
- replace `-d` with correct `-datadir`
- add rerun attempt for failed tests
- some small changes on the go

### CODE REVIEW

File: tests/settings/test_ens_name_purchase.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def keys_screen(main_window) -> KeysView:
 @pytest.mark.parametrize('user_account', [constants.user.user_with_funds])
 @pytest.mark.parametrize('ens_name', [pytest.param(constants.user.ens_user_name)])
 @pytest.mark.skip(reason='https://github.com/status-im/status-desktop/issues/12988')
+
```
Comment (Valentina1133) [inline tests/settings/test_ens_name_purchase.py:35]: I think we don't need this line :)

File: tests/onboarding/test_onboarding_generate_new_keys.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def keys_screen(main_window) -> KeysView:
 @allure.testcase('https://ethstatus.testrail.net/index.php?/cases/view/703010', 'Settings - Sign out & Quit')
 @pytest.mark.case(703421, 703010)
 @pytest.mark.critical
-@pytest.mark.flaky
+
```
Comment (Valentina1133) [inline tests/onboarding/test_onboarding_generate_new_keys.py:35]: And this one too

**Review by Valentina1133 — APPROVED** [review]

File: tests/onboarding/test_onboarding_generate_new_keys.py (hunk 2 above)
Comment (anastasiyaig) [reply tests/onboarding/test_onboarding_generate_new_keys.py:35]: thanks!

**Review by jakubgs — APPROVED** [review]
Comment needs to be indented another 2 spaces.

**Pull request merged**